src/vl-tinymce-link-toolbar.js

```diff
@@ -0,0 +1,114 @@
+import {vlElement, define} from '/node_modules/vl-ui-core/dist/vl-core.js';
+import '/node_modules/vl-ui-modal/dist/vl-modal.js';
+import '/node_modules/vl-ui-form-grid/dist/vl-form-grid.js';
+import '/node_modules/vl-ui-form-message/dist/vl-form-message.js';
+import '/node_modules/vl-ui-input-field/dist/vl-input-field.js';
+import '/node_modules/vl-ui-button/dist/vl-button.js';
+
+export const vlLinkToolbar = {
+  icon: 'link',
+  title: 'Link',
+  tooltip: 'Link',
+  onSetup: () => {
+    console.log('todo');
+  },
+  onAction: () => {
+    // TODO only add modal if it does not exist
+    // it is placed inside here because of application rerender
+    const modal = document.createElement('vl-textarea-modal');
+    const target = tinymce.activeEditor.targetElm;
+    const parent = target.parentElement || target.getRootNode();
+    parent.append(modal);
+    customElements.whenDefined('vl-textarea-modal').then(() => {
+      modal.onSubmit(() => {
+        tinymce.activeEditor.insertContent(`<a target="_blank" href="${modal.url}">${modal.text}</a>`);
+        modal.clear();
+      });
+      modal.open();
+    });
+  },
+};
+
+class VlTextareaModal extends vlElement(HTMLElement) {
+  constructor() {
+    super(`
+      <style>
+        @import '/node_modules/vl-ui-core/dist/style.css';
+        @import '/node_modules/vl-ui-form-grid/dist/style.css';
+        @import '/node_modules/vl-ui-form-message/dist/style.css';
+        @import '/node_modules/vl-ui-input-field/dist/style.css';
+        @import '/node_modules/vl-ui-button/dist/style.css';
+      </style>
+      <vl-modal id="modal-cl" data-title="Link toevoegen" closable>
+        <form id="link-form" slot="content" data-validate-form>
+          <div is="vl-form-grid" is-stacked>
+            <div is="vl-form-column" size="12">
+                <label is="vl-form-label" for="text" data-vl-block>Tekst</label>
+                <input id="text" is="vl-input-field" placeholder="Link" data-vl-block data-required="true" data-vl-error-message="Gelieve een tekst in te vullen" data-vl-error-placeholder="text-error">
```

Comment:
Deze error en die van url heb ik nog niet gezien. Je kan "toevoegen" zonder dat de velden ingevuld zijn. Ook wanneer tekst niet ingevuld is en de link wel en vice versa

Reply:
Je moet (tijdelijk) de #validity branch van vl-ui-input-field gebruiken om deze functionaliteit goed te laten werken.